master/buildbot/status/web/build.py

```diff
@@ -360,23 +360,10 @@ def content(self, req, cxt):
 
             step['link'] = path_to_step(req, s)
             step['text'] = " ".join(s.getText())
-            urls = []
-
-            # TODO: rather than doing this, it'd be easier to refactor logname
-            for url in s.getURLs():
-                urlName = url["name"]
-                urlLink = url["url"]
-                if "results" in url:
-                    if url["results"] in css_classes:
-                        url_dict = dict(logname=urlName, url=urlLink['url'] + codebases_arg,
-                                        results=css_classes[url['results']])
-                    else:
-                        url_dict = dict(logname=urlName, url=urlLink['url'] + codebases_arg)
-                else:
-                    url_dict = dict(logname=urlName, url=urlLink + codebases_arg)
-                urls.append(url_dict)
 
-            step['urls'] = urls
+            step['urls'] = s.getURLs()
+            step['urls'].extend(s.getArtifacts())
+            step['urls'].extend(s.getDependencies())
 
```

Comment:
Didn't we decide that we need to make sure that nothing else is using the output from these two modules before making the change to the actual output? For now, I was just changing the interior code alone and making sure this all works before changing the... er, API.